### PR TITLE
Add 'Wait for' example to asa_command module 

### PR DESCRIPTION
**Pull request by samccann**
##### SUMMARY

Pulls in example from PR41069 with the addition of names for each task, and removes deprecated items.

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

asa_command
##### ANSIBLE VERSION

```paste below
a$ ansible --version
ansible 2.7.0.dev0
  config file = /home/samccann/.ansible.cfg
  configured module search path = [u'/home/samccann/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, May 31 2018, 09:41:32) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]

```

##### ADDITIONAL INFORMATION

Used the following playbook to verify all asa_command examples:

---
- name: "Test all ASA examples"
  hosts: asa
  connection: network_cli
  gather_facts: no

  tasks:

  - name: "Show the ASA version"
    asa_command:
      commands:
        - show version

  - name: "Show ASA drops and memory"
    asa_command:
      commands:
        - show asp drop
        - show memory

  - name: "Send repeat pings and wait for the result to pass 100%"
    asa_command:
      commands:
        - ping 10.8.38.76 repeat 20 size 350
      wait_for:
        - result[0] contains 100
      retries: 2

```paste below
ansible-playbook -i inventory -k all-asa-examples.yml 
SSH password: 
/usr/lib/python2.7/site-packages/requests/__init__.py:91: RequestsDependencyWarning: urllib3 (1.15.1) or chardet (3.0.4) doesn't match a supported version!
  RequestsDependencyWarning)

PLAY [Test all ASA examples] ***************************************************

TASK [Show the ASA version] ****************************************************
ok: [test]

TASK [Show ASA drops and memory] ***********************************************
ok: [test]

TASK [Send repeat pings and wait for the result to pass 100%] ******************
ok: [test]

PLAY RECAP *********************************************************************
test                       : ok=3    changed=0    unreachable=0    failed=0   


```

### CODE REVIEW

File: lib/ansible/modules/network/asa/asa_command.py [hunk 1]
```diff
@@ -68,35 +68,37 @@
 """
 
 EXAMPLES = """
-# Note: examples below use the following provider dict to handle
+# Note: examples below use the following group_vars/asa.yaml file to handle
 #       transport and authentication to the node.
 ---
-vars:
```
Comment (acozine) [inline lib/ansible/modules/network/asa/asa_command.py]: Do you still need this line so Ansible loads lines 74-80 as variables?

File: lib/ansible/modules/network/asa/asa_command.py [hunk 2]
```diff
@@ -68,35 +68,37 @@
...
+ansible_user: ansible
+ansible_become: yes
+ansible_become_method: enable
+ansible_become_pass: cisco
+command_timeout: 60
+
 
 ---
-- asa_command:
+- name: "Show the ASA version"
```
Comment (acozine) [inline lib/ansible/modules/network/asa/asa_command.py]: Hooray for names on all the tasks!!!

File: lib/ansible/modules/network/asa/asa_command.py [hunk 3]
```diff
@@ -68,35 +68,37 @@
...
+  asa_command:
     commands:
       - show version
-    provider: "{{ cli }}"
 
-- asa_command:
+- name: "Show ASA drops and memory"
+  asa_command:
     commands:
       - show asp drop
```
Comment (acozine) [inline lib/ansible/modules/network/asa/asa_command.py:81]: Is `asp` correct? It's not new content, so not really part of the PR, but to me this looks like a typo - should it be `asa`?

**Review by acozine — COMMENTED** [review]
This is a good step forward. I've put a couple of questions/comments on here; overall it looks great.

File: lib/ansible/modules/network/asa/asa_command.py (hunk 1 above)
Comment (samccann) [reply lib/ansible/modules/network/asa/asa_command.py]: This is where I'm not sure if I'm explaining it correctly.  In my test, I have a group_vars/ directory with an asa.yml file that contains that section (lines 74-80).  Then I have a playbook that includes all the examples (starting with line 83).   Do we want to encourage the use of group_vars/ or should I go back to having this all in the playbook itself?

File: lib/ansible/modules/network/asa/asa_command.py (hunk 3 above)
Comment (samccann) [reply lib/ansible/modules/network/asa/asa_command.py:81]: just doublechecked in the cisco command ref - it is show asp drop (accelerated security path)...

File: lib/ansible/modules/network/asa/asa_command.py (hunk 1 above)
Comment (samccann) [reply lib/ansible/modules/network/asa/asa_command.py]: based on sidebar discussion, removing the variables altogether and just keeping these as examples, as most modules use this approach
